src/db_stats.hpp

```diff
@@ -44,16 +44,84 @@ class DbStats
 
     void read_complete(metricq::Duration active_duration, std::size_t data_size);
 
+    void read_failed(metricq::Duration active_duration);
+
     void write_pending();
 
     void write_active(metricq::Duration pending_duration);
 
     void write_complete(metricq::Duration active_duration, std::size_t data_size);
 
+    void write_failed(metricq::Duration active_duration);
+
     void collect();
 
 private:
     class DbStatsImpl;
 
     std::unique_ptr<DbStatsImpl> impl;
+};
+
+class DbStatsReadTransaction
```

Comment:
Oh well. I mean I get the rationale behind this, but can you really sleep at night knowing you just copied this class and changed read to write?